tests/test_data.py

```diff
@@ -620,3 +620,39 @@ def test_delete_dataset_thats_in_a_collection(self, session, network_with_data):
             updated_dataset_ids.append(item.dataset_id)
 
         assert dataset_id not in updated_dataset_ids
+
+class TestiUtilities:
+    """
+        Test hydra's internal utilities relating to data transformations etc.
+    """
+
+    def test_count_levels(self):
+        empty     = {}
+        one_level = {'a':1}
+        two_levels = {'a': {'b': 2}}
+        two_levels_empty = {'a': {'b':{}}}
+
+        assert hb.util.count_levels(empty) == 0
+        assert hb.util.count_levels(one_level) == 1
+        assert hb.util.count_levels(two_levels) == 2
+        assert hb.util.count_levels(two_levels_empty) == 2
+
+    def test_flatten_dict(self):
+        empty     = {}
```

Comment:
These different cases could be done with `@pytest.mark.parametrize`. Minor point for such a simple test.